nexus/src/app/instance.rs

```diff
@@ -774,6 +774,108 @@ impl super::Nexus {
             .map_err(Into::into)
     }
 
+    /// Forcefully stop a running instance, causing its sled-agent to rudely
+    /// terminate its VMM process and unregister the instance.
+    pub(crate) async fn instance_force_terminate(
+        &self,
+        opctx: &OpContext,
+        instance_lookup: &lookup::Instance<'_>,
+    ) -> Result<InstanceAndActiveVmm, InstanceStateChangeError> {
+        let (.., authz_instance) =
+            instance_lookup.lookup_for(authz::Action::Modify).await?;
+
+        let state = self
+            .db_datastore
+            .instance_fetch_with_vmm(opctx, &authz_instance)
+            .await?;
+
+        // Is the instance currently incarnated in a VMM process?
```

Comment:
It's less critical for getting an instance out of a wedged-in-Starting/wedged-in-Stopping state, but I'd envisioned this also terminating the current migration target, if there is one (so that there are no living VMMs consuming resources on behalf of the instance).

Reply:
Mm, yeah, this is a good point, we should probably fetch the target VMM as well, and unregister it too.